### PR TITLE
Extract billy

### PR DESCRIPTION
**Pull request by Serabe**
Billy is a new library directly extracted from go-git. It abstract
several storages systems in a filesystem interface.

More in [github.com/src-d/billy](https://github.com/src-d/billy)

### CODE REVIEW

File: storage/filesystem/storage_test.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"testing"
 
 	"gopkg.in/src-d/go-git.v4/storage/test"
-	"gopkg.in/src-d/go-git.v4/utils/fs/os"
+	"srcd.works/go-billy.v0/os"
```
Comment (smola) [inline storage/filesystem/storage_test.go]: Now that it is an external library, it should go into the third import block.

Comment (Serabe) [inline storage/filesystem/storage_test.go]: I've gone through all the changed files and added it to a third group. When needed, I've grouped other imports as well.

Thank you for the review!

File: README.md [hunk 2]
```diff
@@ -134,11 +134,11 @@ import (
 	"io"
 
 	"gopkg.in/src-d/go-git.v4"
-	"gopkg.in/src-d/go-git.v4/utils/fs"
+	"srcd.works/go-billy.v0/os"
```
Comment (smola) [inline README.md]: grouping? ;)

**Comment by mcuadros** [comment]
rebase?

File: fixtures/fixtures.go [hunk 3]
```diff
@@ -7,12 +7,12 @@ import (
...
 
-	"github.com/alcortesm/tgz"
-
 	"gopkg.in/check.v1"
 	"gopkg.in/src-d/go-git.v4/plumbing"
-	"gopkg.in/src-d/go-git.v4/utils/fs"
-	osfs "gopkg.in/src-d/go-git.v4/utils/fs/os"
+
+	"github.com/alcortesm/tgz"
+	"srcd.works/go-billy.v0"
```
Comment (mcuadros) [inline fixtures/fixtures.go]: please create a tag in go-bily and change the import to a version

**Comment by mcuadros** [comment]
@serabe last rebase

**Comment by Serabe** [comment]
@mcuadros done. Waiting for CI.

**Pull request merged**